homeassistant/components/alert.py

```diff
@@ -0,0 +1,233 @@
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_NOTIFIERS = 'notifiers'
+CONF_REPEAT = 'repeat'
+CONF_SKIP_FIRST = 'skip_first'
+
+DOMAIN = 'alert'
+
+ALERT_SCHEMA = vol.Schema({
+    # pylint: disable=no-value-for-parameter
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_ENTITY_ID): cv.string,
+    vol.Optional(CONF_STATE): cv.string,
+    vol.Required(CONF_REPEAT): cv.positive_int,
+    vol.Optional(CONF_SKIP_FIRST): cv.boolean,
+    vol.Required(CONF_NOTIFIERS): cv.ensure_list})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_ENTITIES): [ALERT_SCHEMA],
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+ALERT_SERVICE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+ALL_ALERTS = {}
+
+
+def is_on(hass, entity_id):
+    """Return if the alert is acknowledged."""
+    return hass.states.is_state(entity_id, STATE_ON)
+
+
+def turn_on(hass, entity_id):
+    """Acknowledge the alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.services.call(DOMAIN, SERVICE_TURN_ON, data)
+
+
+@callback
+def async_turn_on(hass, entity_id):
+    """Async acknowledge the alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.async_add_job(
+        hass.services.async_call(DOMAIN, SERVICE_TURN_ON, data))
+
+
+def turn_off(hass, entity_id):
+    """Reset alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.services.call(DOMAIN, SERVICE_TURN_OFF, data)
+
+
+@callback
+def async_turn_off(hass, entity_id):
+    """Async reset the alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.async_add_job(
+        hass.services.async_call(DOMAIN, SERVICE_TURN_OFF, data))
+
+
+def toggle(hass, entity_id):
+    """Toggle acknowledgement of alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.services.call(DOMAIN, SERVICE_TOGGLE, data)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup alert component."""
+    alerts = config.get(DOMAIN)
+
+    @asyncio.coroutine
+    def async_handle_alert_service(service_call):
+        """Handle calls to alert services."""
+        alert_ids = service.extract_entity_ids(hass, service_call)
+
+        for alert_id in alert_ids:
+            alert = ALL_ALERTS[alert_id]
+            if service_call.service == SERVICE_TURN_ON:
+                yield from alert.async_turn_on()
+            elif service_call.service == SERVICE_TOGGLE:
+                yield from alert.async_toggle()
+            else:
+                yield from alert.async_turn_off()
+
+    for alert in alerts.get(CONF_ENTITIES):
+        entity = yield from hass.loop.run_in_executor(
+            None, Alert, hass, alert[CONF_NAME], alert[CONF_ENTITY_ID],
+            alert.get(CONF_STATE, STATE_ON), alert[CONF_REPEAT],
+            alert.get(CONF_SKIP_FIRST, False),
+            alert[CONF_NOTIFIERS])
+        ALL_ALERTS[entity.entity_id] = entity
+
+    descriptions = {}
```

Comment:
Roger. I just wanted to get the general idea out there before descriptions, documentation, and tests. These will come soon.